### PR TITLE
feat(frontend): change button icon on hover tm-319

### PR DESCRIPTION
**Pull request by KrasKa2**
task-319

### CODE REVIEW

**Review by Vfivew — APPROVED** [review]

**Review by Sytnikovakatya — APPROVED** [review]

**Comment by Oleksandra-Masiuk** [comment]
Please, check out my answer in the chat. Maybe it will be helpful

**Review by ShadowOfTheSpace — COMMENTED** [review]
Can you use here className prop from the Icon component? I think you can create a new prop for the button (iconClassName) and pass it to the Icon component. I think it can be helpful for styling Icon outside of the Button component.

P.S. I saw answer in chat and I think my comment is not helpful for this situation(

File: apps/frontend/src/libs/components/button/button.tsx [hunk 1]
```diff
@@ -58,7 +61,8 @@ const Button: React.FC<Properties> = ({
 					onClick={onClick}
 					type={type}
 				>
-					{icon}
+					<div className={styles["icon"]}>{icon}</div>
+					<div className={styles["hoverIcon"]}>{hoverIcon}</div>
```
Comment (what1s1ove) [inline apps/frontend/src/libs/components/button/button.tsx:65]: we will not add an additional icon here. find a better solution

you can add the classname prop to the Icon component to try to handle this issue

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Comment by KrasKa2** [comment]
> Can you use here className prop from the Icon component? I think you can create a new prop for the button (iconClassName) and pass it to the Icon component. I think it can be helpful for styling Icon outside of the Button component.
> 
> P.S. I saw answer in chat and I think my comment is not helpful for this situation, but if you change icon as Oleksandra said you can style it via iconClassName

@ShadowOfTheSpace your advice works only with Oleksandara's advice))

File: apps/frontend/src/libs/components/button/button.tsx [hunk 2]
```diff
@@ -39,7 +41,9 @@ const Button: React.FC<Properties> = ({
 		isDisabled && styles["disabled"],
 	);
 
-	const icon = iconName ? <Icon name={iconName} /> : null;
+	const icon = iconName ? (
+		<Icon className={iconClassName || ""} name={iconName} />
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/button/button.tsx:45]:
```suggestion
		<Icon className={iconClassName ?? ""} name={iconName} />
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/button/button.tsx (hunk 2 above)
Comment (what1s1ove) [reply apps/frontend/src/libs/components/button/button.tsx:45]: change props in Icon components

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/button/button.tsx (hunk 2 above)
Comment (KrasKa2) [reply apps/frontend/src/libs/components/button/button.tsx]: > change props in Icon components
@what1s1ove  I have done changes suggested by @fshabanov and I don't understand what alse I have to do

**Review by Oleksandra-Masiuk — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**